components/director/internal/domain/runtime/repository_test.go

```diff
@@ -32,13 +32,14 @@ func TestPgRepository_GetByID_ShouldReturnRuntimeModelForRuntimeEntity(t *testin
 	timestamp, err := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
 	require.NoError(t, err)
 
-	sqlxDB, sqlMock := mockDatabase(t)
+	sqlxDB, sqlMock := testdb.MockDatabase(t)
```

Comment:
How the tests should look like for the repository which uses generic repository?

To be honest, if we still need to test all the cases, the benefits of the generic repo are not that significant.

I would like to see a generic test suite for repo that utilizes generic repo. Because writing test cases actually takes most of the time while writing repository.

Consider something like this:

```go
func TestRepository_Exists(t *testing.T) {
labelDefRepo := labeldef.NewRepository(conv)
 ts := genericrepo.NewTestSuite(t, labelDefRepo)
expectedQuery := "select 1 as exists from public.label_definitions where tenant_id=$1 and key=$2"

ts.TestExists(func(ctx context.Context) bool, error { 
    return labelDefRepo.Exists(ctx, "anything", "anything")
}, expectedQuery, "anything", "anything") // it will do all cases for exists
}
```

which would test all cases for the Exists method.

Similar approach I would see for Delete:

```go
func TestRepository_Delete(t *testing.T) {
key := ...
objID := ...
objType := ...
tnt := ...

labelRepo := label.NewRepository(conv)
 ts := genericrepo.NewTestSuite(t, labelRepo)
expectedQuery := "DELETE FROM "public"."labels" WHERE "key" = $1 AND "runtime_id" = $2 AND "tenant_id" = $3"

ts.TestDelete(func(ctx context.Context) error { 
    return repo.Delete(ctx, tnt, objType, objID, key)
}, expectedQuery, tnt, objType, objID, key) // it will do all cases for delete
}
```

I believe we can do it for Get and List operations as well. We just need to specify proper test suite method prototype. For example, we could pass function which constructs an repo object for test purposes (to be able to specify different mock converters).